src/udev/udevd.c

```diff
@@ -1675,12 +1675,16 @@ int main(int argc, char *argv[]) {
 
         if (arg_children_max == 0) {
                 cpu_set_t cpu_set;
+                unsigned long mem_limit;
 
                 arg_children_max = 8;
 
                 if (sched_getaffinity(0, sizeof(cpu_set), &cpu_set) == 0)
                         arg_children_max += CPU_COUNT(&cpu_set) * 2;
 
+                mem_limit = physical_memory() / (128LU*1024*1024);
+                arg_children_max = MAX(10U, MIN(arg_children_max, mem_limit));
```

Comment:
Hmm, but do we actually want at least 10 workers? If I have a tiny machine with 256MB or RAM, and a single CPU, wouldn't it be better to have ~4 workers?

Reply:
I was mainly concerned about the upper limit. Therefore I wrote the patch such that it doesn't change the lower limit. What lower limit would you suggest? At SUSE, we've had bug reports that udev processing on small systems was much too slow due to the limited number of workers.

Reply:
OK. So let's keep things as they are and just introduce this one change now.